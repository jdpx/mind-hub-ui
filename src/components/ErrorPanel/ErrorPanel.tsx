import React from 'react'

import errorIcon from '../../assets/error-icon.svg'

import './ErrorPanel.scss'

interface Props {
    height?: number
    message?: string
    testid?: string
}

const defaultErrorMessage = 'Opps something went wrong. Please try again later'

export default function ErrorPanel({
    height = 200,
    message = defaultErrorMessage,
    testid = 'error-panel',
}: Props) {
    return (
        <div className="error-panel" data-testid={testid}>
            <img
                src={errorIcon}
                alt="Opps something went wrong"
                height={height}
                data-testid={`${testid}-img`}
            />
            <h1 data-testid={`${testid}-message`}>{message}</h1>
        </div>
    )
}
