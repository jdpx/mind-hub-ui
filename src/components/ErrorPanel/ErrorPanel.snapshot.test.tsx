import React from 'react'
import { render } from '@testing-library/react'
import ErrorPanel from './ErrorPanel'

describe('ErrorPanel Snapshots', () => {
    it('renders error panel', () => {
        const { asFragment } = render(<ErrorPanel />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders error panel with custom message', () => {
        const { asFragment } = render(<ErrorPanel message="An error occurred" />)
        expect(asFragment()).toMatchSnapshot()
    })
})
