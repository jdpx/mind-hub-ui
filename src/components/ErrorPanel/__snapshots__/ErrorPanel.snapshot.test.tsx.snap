// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPanel Snapshots renders error panel 1`] = `
<DocumentFragment>
  <div
    class="error-panel"
    data-testid="error-panel"
  >
    <img
      alt="Opps something went wrong"
      data-testid="error-panel-img"
      height="200"
      src="error-icon.svg"
    />
    <h1
      data-testid="error-panel-message"
    >
      Opps something went wrong. Please try again later
    </h1>
  </div>
</DocumentFragment>
`;

exports[`ErrorPanel Snapshots renders error panel with custom message 1`] = `
<DocumentFragment>
  <div
    class="error-panel"
    data-testid="error-panel"
  >
    <img
      alt="Opps something went wrong"
      data-testid="error-panel-img"
      height="200"
      src="error-icon.svg"
    />
    <h1
      data-testid="error-panel-message"
    >
      An error occurred
    </h1>
  </div>
</DocumentFragment>
`;
